pkg/packageinstall/packageinstall.go

```diff
@@ -74,14 +75,28 @@ func (pi *PackageInstallCR) Reconcile() (reconcile.Result, error) {
 	}
 
 	// Always update status
-	statusErr := pi.updateStatus()
+	statusErr := pi.UpdateStatus()
 	if statusErr != nil {
 		return reconcile.Result{Requeue: true}, statusErr
 	}
 
 	return result, err
 }
 
+// Validate returns an aggregated error if any fields on the underlying PackageInstall are invalid
+func (pi *PackageInstallCR) Validate() error {
+	var allErrs field.ErrorList
+	for i, value := range pi.model.Spec.Values {
```

Comment:
if we're just checking SecretRef this feels like the right approach,  but given the slightly aspirational comment description "...any fields..." I was getting ready for some more generalized like, openAPI style validation.     Then i  confused myself b/c in https://github.com/vmware-tanzu/carvel-kapp-controller/blob/develop/config/crds.yml it looks like SecretRef is a field on Package, not PackageInstall.   What do you think?

Reply:
I see it on the PKGI @joe-kimmel-vmw https://carvel.dev/kapp-controller/docs/v0.34.0/packaging/#package-install

`spec.values` near the bottom

Reply:
🤦 oh yeah, [ here it is](https://github.com/vmware-tanzu/carvel-kapp-controller/blob/develop/config/crds.yml#L1274)

Reply:
Yeah, I think we're doing some validation mixed into other places right now, we don't have a centralized pattern. I was hoping this might be good a good pattern we could spread to other places.